Update node modules to latest revisions

TheEgghead27 opened the pull request:
This pull request aims to bring the StuyActivities client up to speed with the latest updates to React and Material UI (now rebranded as MUI).

Many breaking changes have been added to our dependencies, so there may have been undetected bugs and regressions in the updated code.
However (at the time of writing), updating and migrating is not 100% complete, with some packages still being somewhat out of date.

Due to [`@mui/styles`, the legacy JSS system, not supporting React 18](https://mui.com/system/styles/basics/), many CSS declarations need to be migrated, with their selectors altered to prevent [bugs with inheritance and other properties](1e1473decd33ec8679f8e3ba8c493577388d2d54).

As a pleasant side effect, most of the `npm audit` vulnerabilities, save for [the erroneous ones](https://github.com/facebook/create-react-app/issues/11174), are simply resolved by updating our dependencies.

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@tinymce/tinymce-react": "^3.14.0",
+		"apollo-upload-client": "^17.0.0",
+		"diff": "^5.1.0",
+		"emoji-picker-react": "^3.6.1",
```
Resolves/Conflicts with #596

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@mui/material": "^5.10.1",
+		"@mui/styles": "^5.9.3",
+		"@mui/x-date-pickers": "^5.0.0-beta.6",
+		"@testing-library/jest-dom": "^5.16.5",
```
Resolves/Conflicts with #595

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@testing-library/jest-dom": "^5.16.5",
+		"@testing-library/react": "^12.1.5",
+		"@testing-library/user-event": "^12.1.10",
+		"@tinymce/tinymce-react": "^3.14.0",
```
Conflicts with (**but does not resolve!**) #594

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@apollo/client": "^3.6.9",
+		"@emotion/react": "^11.10.0",
+		"@emotion/styled": "^11.10.0",
+		"@fullcalendar/daygrid": "^5.11.2",
```
Resolves/Conflicts with #592

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@fullcalendar/daygrid": "^5.11.2",
+		"@fullcalendar/list": "^5.11.2",
+		"@fullcalendar/react": "^5.11.2",
+		"@honeybadger-io/react": "2.1.0",
```
Resolves/Conflicts with #591

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
-		"react-dom": "^16.14.0",
-		"react-dropzone": "^11.3.2",
-		"react-ga": "^3.3.0",
+		"notistack": "^2.0.5",
```
Resolves/Conflicts with #590

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"web-vitals": "^2.1.4"
+	},
+	"devDependencies": {
+		"react-scripts": "5.0.1"
```
Resolves/Conflicts with #588

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
-		"react-router-dom": "^5.2.0",
-		"react-scripts": "4.0.3",
+		"react-multi-carousel": "^2.8.2",
+		"react-router-dom": "^5.3.3",
```
Conflicts with (**but does not resolve!**) #566

TheEgghead27 commented on package.json:
```diff
@@ -3,48 +3,51 @@
+		"@testing-library/react": "^12.1.5",
+		"@testing-library/user-event": "^12.1.10",
+		"@tinymce/tinymce-react": "^3.14.0",
+		"apollo-upload-client": "^17.0.0",
```
Resolves/Conflicts with #540

frank-cywong commented on package.json:
```diff
@@ -75,5 +86,17 @@
+		"url": "git+https://github.com/stuysu/stuyactivities.org.git"
+	},
+	"author": "",
+	"license": "ISC",
```
@TheEgghead27 Did we ever choose a license properly for this? I don't think we did

TheEgghead27 replied:
Ah that's the default npm template, should be able to remove that when I get home 👍

TheEgghead27 commented:
Implemented the dark mode toggle and rebased this PR upon the latest `master` to address any merge conflicts that have arose. Should be ready for review/testing (as soon as we authorize CORS in the Google Cloud OAuth dashboard),

frank-cywong commented on src/comps/context/ThemeProvider.js:
```diff
@@ -1,58 +1,128 @@
+						contrastText: "#fff"
+					},
+					button: {
+						main: prefersDarkMode ? "rgba(255, 255, 255, 0.87)" : "rgba(0, 0, 0, 0.87)"
```
Something here seems to be a bit off since buttons are displayed as if they were in dark mode even in light mode

Also the calendar (and text) seem slightly bigger than before but I can't pinpoint why

![ThingAN2213](https://user-images.githubusercontent.com/90653148/192077574-32a9b06a-365a-432d-a55d-c3c71f0b79f6.png)

TheEgghead27 replied:
The fullcalendar components use `em` for styling (baked into the library), which indirectly reference the `<body>`'s `font-size: 0.875rem` specification. The updated MUI theme changes the `<body>` font size to 1rem at `htmlFontSize: 16`, rather than `0.875rem` as it was previously. To fix it, a parent div has been added that gives `<Calendar>` the expected parent fontSize.

frank-cywong commented on src/comps/Calendar.js:
```diff
@@ -1,30 +1,24 @@
 import React from "react";
-import { makeStyles } from "@material-ui/core";
+import Box from "@mui/material/Box";
 import FullCalendar from "@fullcalendar/react";
```
Also another calendar thing, is it possible to change the colour of the current day in dark mode?
![ThingAN2214](https://user-images.githubusercontent.com/90653148/192077668-3316f3bb-2a7e-43f9-b237-eee2f7e66983.png)

TheEgghead27 replied:
Yup! Fullcalendar takes another CSS variable to customize the coloration of the today highlight, and I have a theme-aware use of it in 89473539597d831bdeb649ebd22b2549d54d3a82. Let me know what you think about the current color I set (`rgba(255, 255, 255, 0.05)`).

frank-cywong commented on src/index.css:
```diff
@@ -49,13 +49,6 @@ p {
 	}
 }
 
-:root {
```
This might be related to the calendar colour issue mentioned elsewhere

TheEgghead27 replied:
Yup! I reverted this change in commit `01e8caf2f53a13cc7628337caf7f36b5537acba0` to resolve it.

frank-cywong reviewed (changes requested):
Done the stuff I could address so far with what I can test without CORS-based login working

frank-cywong commented:
Tested GUI code that needs logging in by doing some cookie manipulation to log in
Seems mostly good, and I doubt any backend logic has changed (hopefully)

One very tiny nitpick:
![ThingAN2230](https://user-images.githubusercontent.com/90653148/193722299-8c0e7e98-18f8-47a5-a9ae-fb38cad5a42f.png)

The fancy TinyEditor used for things like meetings doesn't have a dark mode currently, if you look into it and see that it's not easy to add it's fine if you don't add it, really good work :+1: and I'll merge this PR probably this weekend so I can troubleshoot if any issues arise

TheEgghead27 commented:
Looked into TinyEditor dark modes, and [it is possible](https://www.tiny.cloud/blog/dark-mode-tinymce-rich-text-editor/), however the theme appears to obscure the placeholder text.
![image](https://user-images.githubusercontent.com/45245358/193951368-c3a06152-d10d-4987-b041-b42d2e76c271.png)

TheEgghead27 commented:
Alternatively, it's possible to only apply dark mode to the UI, not the contents, but I'm not sure if it looks better or worse
![image](https://user-images.githubusercontent.com/45245358/193954304-7994a64d-3eef-48e3-ba02-c3cf91f05311.png)

frank-cywong commented:
> Looked into TinyEditor dark modes, and [it is possible](https://www.tiny.cloud/blog/dark-mode-tinymce-rich-text-editor/), however the theme appears to obscure the placeholder text. ![image](https://user-images.githubusercontent.com/45245358/193951368-c3a06152-d10d-4987-b041-b42d2e76c271.png)
> 
> If this is acceptable, I can implement it.

Does this work perhaps or is this outdated? https://stackoverflow.com/questions/64867107/tinymce-how-to-change-the-placeholder-color
https://www.tiny.cloud/blog/tinymce-placeholder-text/

If it doesn't work the latter option or just leaving it bright would suffice as well and we can deal with this in the future
